src/libstd/sys/unix/fd.rs

```diff
@@ -50,6 +50,16 @@ impl FileDesc {
         (&mut me).read_to_end(buf)
     }
 
+    pub fn read_at(&self, buf: &mut [u8], offset: u64) -> io::Result<usize> {
+        let ret = cvt(unsafe {
+            libc::pread(self.fd,
+                        buf.as_mut_ptr() as *mut c_void,
+                        buf.len(),
+                        offset as off_t)
```

Comment:
This truncation can be significant, right? `off_t` may be 32-bit on some platforms?

I think there's also `pread64` for this on Linux at least